command/resource/client/grpc-client.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+type GRPCClient struct {
+	Client pbresource.ResourceServiceClient
+	Config *GRPCConfig
+	Conn   *grpc.ClientConn
+}
+
+func NewGRPCClient(config *GRPCConfig) (*GRPCClient, error) {
+	conn, err := dial(config)
+	if err != nil {
+		return nil, fmt.Errorf("**** error dialing grpc: %+v", err)
+	}
+	return &GRPCClient{
+		Client: pbresource.NewResourceServiceClient(conn),
+		Config: config,
+		Conn:   conn,
+	}, nil
+}
+
+func dial(c *GRPCConfig) (*grpc.ClientConn, error) {
+	dialCtx := context.Background()
```

Comment:
I think we should have a way to cancel the context here. Maybe use a cancel context and keep the cancel channel in the Client?

Reply:
I reviewed the situation and I don't think I need the context here so I'll dial the server directly.